pcr_profile.go

```diff
@@ -0,0 +1,162 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package secboot
+
+import (
+	"fmt"
+
+	"github.com/chrisccoulson/go-tpm2"
+
+	"golang.org/x/xerrors"
+)
+
+// pcrValuesList is a list of PCR value combinations computed from PCRProtectionProfile.
+type pcrValuesList []tpm2.PCRValues
+
+// setValue sets the specified PCR to the supplied value for all branches.
+func (l pcrValuesList) setValue(alg tpm2.HashAlgorithmId, pcr int, value tpm2.Digest) {
+	for _, v := range l {
+		v.SetValue(pcr, alg, value)
+	}
+}
+
+// extendValue extends the specified PCR with the supplied value for all branches.
+func (l pcrValuesList) extendValue(alg tpm2.HashAlgorithmId, pcr int, value tpm2.Digest) {
+	for _, v := range l {
+		if _, ok := v[alg]; !ok {
```

Comment:
the two if feel like the should be a helper on PCRValues,  not a fan of the partial encapsulation going here